### PR TITLE
tests: run snap-set-core-config on all core devices

### PR DESCRIPTION
**Pull request by mvo5**
The snapctl-configure-core has a bit of a misleading name. It is
really about testing all the "snap set core " options. So this
commit renames/simplifies the test (we don't need a fake
bind mount anymore to test this). The new name is
snap-set-core-config.

Fix build on top of https://github.com/snapcore/snapd/pull/7766

### CODE REVIEW

File: overlord/configstate/configcore/proxy.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func etcEnvironment() string {
 func updateEtcEnvironmentConfig(path string, config map[string]string) error {
 	f, err := os.OpenFile(path, os.O_RDWR|os.O_CREATE, 0644)
 	if err != nil {
-		return nil
+		return err
```
Comment (zyga) [inline overlord/configstate/configcore/proxy.go]: Good catch!

**Comment by zyga** [comment]
This fails and needs the core18 change to land first:
```
error: cannot perform the following tasks:
- Run configure hook of "core" snap (run hook "configure": open /etc/environment: read-only file system)
```

File: tests/main/snap-set-core-config/task.yaml [hunk 2]
```diff
@@ -1,18 +1,9 @@
...
-    #!/bin/sh
-    snapctl internal configure-core
-    EOF
-    chmod +x new-configure-core
-    mount -o bind new-configure-core /snap/core/current/meta/hooks/configure
+systems: [ubuntu-core-*]
 
 restore: |
-    umount /snap/core/current/meta/hooks/configure
     systemctl enable rsyslog.service
```
Comment (sergiocazzolato) [inline tests/main/snap-set-core-config/task.yaml]: rsyslog.service not defined on core18

The test is failing with:
+ systemctl status rsyslog.service
Unit rsyslog.service could not be found.
grep error: pattern not found, got:

**Comment by pedronis** [comment]
In a follow up we should consider switching to the more modern "set system"

**Pull request merged**